kernel/src/utilities/leasable_buffer.rs

```diff
@@ -37,18 +37,32 @@ use core::ops::{Bound, Range, RangeBounds};
 use core::ops::{Index, IndexMut};
 use core::slice::SliceIndex;
 
+/// Leasable buffer which can be used to pass a section of a larger mutable
+/// buffer but still get the entire buffer back in a callback
+#[derive(Debug, PartialEq)]
+pub struct LeasableMutableBuffer<'a, T> {
+    internal: &'a mut [T],
+    active_range: Range<usize>,
+}
+
 /// Leasable Buffer which can be used to pass a section of a larger buffer but still
 /// get the entire buffer back in a callback
+#[derive(Debug, PartialEq)]
 pub struct LeasableBuffer<'a, T> {
-    internal: &'a mut [T],
+    internal: &'a [T],
     active_range: Range<usize>,
 }
 
-impl<'a, T> LeasableBuffer<'a, T> {
+pub enum LeasableBufferDynamic<'a, T> {
+    Immutable(LeasableBuffer<'a, T>),
+    Mutable(LeasableMutableBuffer<'a, T>),
+}
```

Comment:
What about re-using `MutImutBuffer` here and storing the `active_range` separately?

That way we can keep all `Immutable`/`Mutable` buffers inside the one type. I feel that might simplify things if we ever migrate t a new way of doing it.

I has the obvious downside of converting to/from `LeasableBuffer` and `LeasableMutableBuffer` is a little clunkier, but it shouldn't be too bad

Reply:
I found it easier to keep the 3 values coupled, rather than combining them in a LeasableBuffer, breaking them apart in separate fields, then regenerating the LeasableBuffer on the callback.